tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c

```diff
@@ -115,20 +115,31 @@ static struct wdt_timeout_cfg m_cfg_wdt0;
 static struct wdt_timeout_cfg m_cfg_wdt1;
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32F7X) || defined(CONFIG_SOC_SERIES_STM32H7X)
+/* STM32H7 and STM32F7 guarantee last write RAM retention over reset,
+ * only for 64bits
+ * See details in Application Note
+ * https://www.st.com/resource/en/application_note/dm00623136-error-correction-code-ecc-management-for-internal-memories-protection-on-stm32h7-series-stmicroelectronics.pdf
```

Comment:
I think it's better to just reference application note number / reference, URLs change frequently.

Reply:
Done